Source/Parallelization/WarpXRegrid.cpp

```diff
@@ -228,10 +248,112 @@ WarpX::RemakeLevel (int lev, Real time, const BoxArray& ba, const DistributionMa
             costs[lev]->setVal(0.0);
         }
 
+    if (costs_heuristic[lev] != nullptr) {
+            costs_heuristic[lev].reset(new MultiFab(costs_heuristic[lev]->boxArray(), dm, 1, 0));
+            costs_heuristic[lev]->setVal(0.0);
+        }
+
         SetDistributionMap(lev, dm);
     }
     else
     {
         amrex::Abort("RemakeLevel: to be implemented");
     }
 }
+
+void
+WarpX::ComputeCostsHeuristic (int lev)
+{
+    MultiFab* cost = WarpX::getCostsHeuristic(lev);
+    auto & mypc = WarpX::GetInstance().GetPartContainer();
+    auto nSpecies = mypc.nSpecies();
+    const IntVect& ng = current_fp[lev][0]->nGrowVect();
```

Comment:
I do not think we can use that: different `MultiFabs` have different number of guard cells.

Reply:
Changed to match the suggestion for 'comment on lines 298 to 334'.